### PR TITLE
Add mtt calloc() implementation

### PR DESCRIPTION
**Pull request by PatKamin**
### Description

Requires #805 
### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [ ] Other

### Checklist

- [ ] Code compiles without errors
- [ ] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [ ] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/803)

### CODE REVIEW

**Review by bratpiorka — APPROVED** [review]
> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/803)* status: all files reviewed, 1 unresolved discussion (waiting on @maciekpac and @PatKamin)

___
*[-- commits, line 5 at r3](https://reviewable.io/reviews/memkind/memkind/803#-N00MgYr0pFOX1Y8awrB):*
nit: ok but could you change this commit info to "mtt calloc parellel test" or similar?

**Review by maciekpac — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, 4 of 4 files at r5, 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/803)* status: all files reviewed, 1 unresolved discussion (waiting on @PatKamin)

___
*[test/memkind_memtier_data_movement_test.cpp, line 1834 at r6](https://reviewable.io/reviews/memkind/memkind/803#-N00y6GdD93gpODYZEm-:-N00y6GdD93gpODYZEm0:b-bgeuez) ([raw file](https://github.com/memkind/memkind/blob/1bfbeec1a6ca76a847813b6d8f281558e24df2f6/test/memkind_memtier_data_movement_test.cpp#L1834)):*
> ```cpp
>     MTTAllocator mtt_allocator;
> 
>     mtt_allocator_create(&mtt_allocator, &limits);
> ```

is mtt_allocator_destroy called?

**Review by PatKamin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/803)* status: 2 of 5 files reviewed, 1 unresolved discussion (waiting on @maciekpac)

___
*[test/memkind_memtier_data_movement_test.cpp, line 1834 at r6](https://reviewable.io/reviews/memkind/memkind/803#-N00y6GdD93gpODYZEm-:-N04foHhDPp83LM52rUr:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/1bfbeec1a6ca76a847813b6d8f281558e24df2f6/test/memkind_memtier_data_movement_test.cpp#L1834)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

is mtt_allocator_destroy called?
</blockquote></details>

Done.

**Review by bratpiorka — APPROVED** [review]
> Reviewed 1 of 4 files at r5, 3 of 3 files at r7, 2 of 2 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/803)* status: all files reviewed, 1 unresolved discussion (waiting on @maciekpac)

File: src/mtt_allocator.c [hunk 1]
```diff
@@ -384,6 +384,19 @@ MEMKIND_EXPORT void *mtt_allocator_malloc(MTTAllocator *mtt_allocator,
...
 }
 
+MEMKIND_EXPORT void *mtt_allocator_calloc(MTTAllocator *mtt_allocator,
+                                          size_t num, size_t size)
+{
+    size_t alloc_size = num * size;
+    void *ret = mtt_allocator_malloc(mtt_allocator, alloc_size);
+    if (ret == NULL) {
+        log_fatal("Allocation failed!");
+        abort();
```
Comment (maciekpac) [inline src/mtt_allocator.c:394]: I'm not sure if we should abort here - if mtt_allocator_malloc did not abort, why should we ? We can just propagate the null value, I wouldn't even add additional logs here.

**Review by PatKamin — COMMENTED** [review]
> Reviewed 1 of 3 files at r9.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/803)* status: 3 of 5 files reviewed, 2 unresolved discussions (waiting on @bratpiorka and @maciekpac)

___
*[src/mtt_allocator.c, line 394 at r8](https://reviewable.io/reviews/memkind/memkind/803#-N-gREKtYWbfdOjoJGD3-r8-394:-N063mCdEONdHGy70zXL:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/229f2aa081563a8868dd548e3339a39fc5774d38/src/mtt_allocator.c#L394)):*
<details><summary><i>Previously, maciekpac wrote…</i></summary><blockquote>

I'm not sure if we should abort here - if mtt_allocator_malloc did not abort, why should we ? We can just propagate the null value, I wouldn't even add additional logs here. 
</blockquote></details>

Done.

**Review by maciekpac — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/803)* status: 3 of 5 files reviewed, 1 unresolved discussion (waiting on @bratpiorka and @maciekpac)

**Review by bratpiorka — APPROVED** [review]
> Reviewed 3 of 3 files at r9, 2 of 2 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/803)* status: all files reviewed, 1 unresolved discussion (waiting on @maciekpac)

**Review by maciekpac — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/803)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @PatKamin)

**Pull request merged**